sui-execution/latest/sui-move-natives/src/test_scenario.rs

```diff
@@ -95,6 +95,7 @@ pub fn end_transaction(
 ) -> PartialVMResult<NativeResult> {
     assert!(ty_args.is_empty());
     assert!(args.is_empty());
+    // Remove the object from storage. We should never hit this scenario either.
```

Comment:
I don't follow this comment (maybe this was a debug comment for you, in which case feel free to ignore my attempts to understand it 😅):

- "the object" -- which object?
- "this scenario" -- are we talking about the `incorrect_shared_or_imm_handling` logic below, or something else?
- "never" -- assuming we're talking about the code block that immediately follows, it seems like it's possible to hit this case, if the client mis-handles a shared or immutable type.

...I'm so confused.

Reply:
I am too lol I'm going to remove as it makes no sense to me either...